From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 8 May 2017 11:36:56 +0200
Subject: [PATCH] {PRINTF} drivers/staging/typec: add __printf attribute to
 logging function

Commit e79e01254d02ff7c460851c773135db35bd2abb6 ("staging: typec: Add
__printf verification") seems to have added some __printf attributes
too, in 4.14-rc1
---
 drivers/staging/typec/fusb302/fusb302.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/staging/typec/fusb302/fusb302.c b/drivers/staging/typec/fusb302/fusb302.c
index fc6a3cf74eb3..832a6ba2108b 100644
--- a/drivers/staging/typec/fusb302/fusb302.c
+++ b/drivers/staging/typec/fusb302/fusb302.c
@@ -153,7 +153,7 @@ static bool fusb302_log_full(struct fusb302_chip *chip)
 		(chip->logbuffer_head + 1) % LOG_BUFFER_ENTRIES;
 }
 
-static void _fusb302_log(struct fusb302_chip *chip, const char *fmt,
+static __printf(2, 0) void _fusb302_log(struct fusb302_chip *chip, const char *fmt,
 			 va_list args)
 {
 	char tmpbuffer[LOG_BUFFER_ENTRY_SIZE];
@@ -200,7 +200,7 @@ static void _fusb302_log(struct fusb302_chip *chip, const char *fmt,
 	mutex_unlock(&chip->logbuffer_lock);
 }
 
-static void fusb302_log(struct fusb302_chip *chip, const char *fmt, ...)
+static __printf(2, 3) void fusb302_log(struct fusb302_chip *chip, const char *fmt, ...)
 {
 	va_list args;
 
@@ -264,7 +264,7 @@ static void fusb302_debugfs_exit(struct fusb302_chip *chip)
 
 #else
 
-static void fusb302_log(const struct fusb302_chip *chip,
+static __printf(2, 3) void fusb302_log(const struct fusb302_chip *chip,
 			const char *fmt, ...) { }
 static int fusb302_debugfs_init(const struct fusb302_chip *chip) { return 0; }
 static void fusb302_debugfs_exit(const struct fusb302_chip *chip) { }
-- 
